File: quickstart/MyStrategy.cs
using AiCup22.Model;
using System.Collections.Generic;

namespace AiCup22
{
    public class MyStrategy
    {
        public MyStrategy(AiCup22.Model.Constants constants) {}
        public AiCup22.Model.Order GetOrder(AiCup22.Model.Game game, DebugInterface debugInterface)
        {
            Order order = new Order();
            IDictionary<int, UnitOrder> orders = new Dictionary<int, UnitOrder>();
            ActionOrder aim = new ActionOrder.Aim(true);
           
            foreach(Unit unit in game.Units)
            {
                if (unit.PlayerId != game.MyId)
                {
                    continue;
                }
                orders.Add(unit.Id, new UnitOrder(
                new Vec2(-unit.Position.X, -unit.Position.Y),
                new Vec2(-unit.Position.Y, unit.Position.X),
                new ActionOrder.Aim(true)));
            }

            return new Order(orders);
        }
        public void DebugUpdate(int displayedTick, DebugInterface debugInterface) {}
        public void Finish() {}
    }
}

File: quickstart/my_strategy.rs
use crate::debug_interface::DebugInterface;
use ai_cup_22::*;
use ai_cup_22::model::UnitOrder;
use std::collections::HashMap;

pub struct MyStrategy {}

impl MyStrategy {
    pub fn new(_constants: model::Constants) -> Self {
        Self {}
    }
    pub fn get_order(
        &mut self,
        game: &model::Game,
        _debug_interface: Option<&mut DebugInterface>,
    ) -> model::Order {
        let mut orders: HashMap<i32, UnitOrder> = HashMap::new();

        for unit in &game.units {
            if unit.player_id != game.my_id {
                continue;
            }

            orders.insert(
                unit.id,
                UnitOrder {
                    target_velocity: model::Vec2 {
                        x: -unit.position.x,
                        y: -unit.position.y,
                    },
                    target_direction: model::Vec2 {
                        x: -unit.direction.y,
                        y: unit.direction.x,
                    },
                    action: Option::from(model::ActionOrder::Aim { shoot: true }),
                },
            );
        }

        model::Order {
            unit_orders: orders,
        }
    }
    pub fn debug_update(&mut self, _displayed_tick: i32, _debug_interface: &mut DebugInterface) {}
    pub fn finish(&mut self) {}
}
